src/radical/saga/adaptors/pbspro/pbsprojob.py

```diff
@@ -161,85 +161,37 @@ def _script_generator(url, logger, jd, ppn, gres, version, is_cray=False,
-
-    if jd.environment:
-        pbs_params += "#PBS -v %s\n" % \
-                ','.join (["%s=%s" % (k,v)
-                           for k,v in jd.environment.items()])
-
-    # apparently this doesn't work with older PBS installations
-    #    if jd.working_directory:
-    #        pbs_params += "#PBS -d %s \n" % jd.working_directory
-    #
-    # a workaround is to do an explicit 'cd'
-    if jd.working_directory:
-        workdir_directives  = 'export    PBS_O_WORKDIR=%s \n' % jd.working_directory
-        workdir_directives += 'mkdir -p  %s\n' % jd.working_directory
-        workdir_directives += 'cd        %s\n' % jd.working_directory
-    else:
-        workdir_directives = ''
+        pbs_params += '#PBS -N %s\n' % jd.name
 
     if jd.output:
-        # if working directory is set, we want stdout to end up in
-        # the working directory as well, unless it containes a specific
-        # path name.
-        if jd.working_directory:
-            if os.path.isabs(jd.output):
-                pbs_params += "#PBS -o %s \n" % jd.output
-            else:
-                # user provided a relative path for STDOUT. in this case
-                # we prepend the workind directory path before passing
-                # it on to PBS
-                pbs_params += "#PBS -o %s/%s \n" % (jd.working_directory, jd.output)
-        else:
-            pbs_params += "#PBS -o %s \n" % jd.output
+        stdout_dir = ''
+        if jd.working_directory and not os.path.isabs(jd.output):
+            # user provided a relative path for STDOUT
+            stdout_dir = '%s/' % jd.working_directory
+        pbs_params += '#PBS -o %s%s\n' % (stdout_dir, jd.output)
 
     if jd.error:
-        # if working directory is set, we want stderr to end up in
-        # the working directory as well, unless it contains a specific
-        # path name.
-        if jd.working_directory:
-            if os.path.isabs(jd.error):
-                pbs_params += "#PBS -e %s \n" % jd.error
-            else:
-                # user provided a realtive path for STDERR. in this case
-                # we prepend the workind directory path before passing
-                # it on to PBS
-                pbs_params += "#PBS -e %s/%s \n" % (jd.working_directory, jd.error)
-        else:
-            pbs_params += "#PBS -e %s \n" % jd.error
-
+        stderr_dir = ''
+        if jd.working_directory and not os.path.isabs(jd.error):
+            # user provided a relative path for STDERR
+            stderr_dir = '%s/' % jd.working_directory
+        pbs_params += '#PBS -e %s%s\n' % (stderr_dir, jd.error)
 
     if jd.wall_time_limit:
-        hours = int(jd.wall_time_limit / 60)
-        minutes = jd.wall_time_limit % 60
-        pbs_params += "#PBS -l walltime=%s:%s:00 \n" \
-            % (str(hours), str(minutes))
+        pbs_params += '#PBS -l walltime=%s:%s:00\n' \
+            % (str(int(jd.wall_time_limit / 60)), str(jd.wall_time_limit % 60))
```

Comment:
The `str()` conversion is enforced by `%s` anyway - I would suggest to remove it for clarity.